### ISSUE TITLE
Cannot find shiny port number

### ISSUE THREAD

**Issue by jstockwin** [rstudio/shinytest#99]
Sorry if I am doing something silly, but I am currently experiencing the following error:
```
> testApp("/home/jake/Development/TransmissibilityEstimator")
====== Running mytests.R ======
Error in sd_startShiny(self, private, path) : 
 Cannot find shiny port number. Error: Error in shinyAppDir(x) : App dir must contain either app.R orserver.R.Calls: <Anonymous> ... as.shiny.appobj -> as.shiny.appobj.character ->shinyAppDirExecution halted
```

The error about not containing `app.R` is simply not true. (I previously had a `server.R` and a `ui,R` which I think should work, but I combined them into one `app.R` just in case, and this didn't solve the issue). 

I can run `shiny::shinyAppDir("/home/jake/Development/TransmissibilityEstimator")` without a problem,  which is strange. I also have my working directory (in R) set to where the app is, so running `testApp()` feels like it should also work, but gives the same error. `recordTest()` worked fine, and surely also starts the app..?

I'm obviously hitting [this](https://github.com/rstudio/shinytest/blob/a8a37d8e7365099a31d96b9be4f2a87ef29bcc7f/R/initialize.R#L123), but I can't work out why. When running `shinyAppDir` I get `Listening on http://127.0.0.1:6275` as expected (and as your code is looking for), but I guess it's not getting this far.

Any ideas much appreciated.

**Comment by wch** [comment]
Hm, that does sound strange. There's no chance that the R process in the VM could start a process in the host.

Is the directory with the app shared from the host to the VM? If so, I could imagine that capitalization of the files (specifically, .r vs .R) could matter, since Windows usually has a case-insensitive filesystem, but Linux expects a case-sensitive one.

**Comment by jstockwin** [comment]
No, nothing is shared. I've got a separate clone on the VM, so can't be
anything like that.

**Comment by KZARCA** [comment]
I have a similar issue. I'm still investigating, but it seems that it's related to the global.R file. Do you have this file in your project?

**Comment by grantirv** [comment]
I have the same issue.

You can replicate by just putting a `Sys.sleep(10)` in global.R for a simple app that works otherwise.
Setting the `loadTimeout` parameter for `ShinyDriver$new` has no effect.
Looking at the code, this because the parameter is not passed down to the `sd_startShiny` function. That function gives up trying to read the port after a fixed 10 seconds. Should this not rather be linked to the `loadTimeout` parameter?

**Comment by michaelhogersosis** [comment]
Hi there, we had the same issue, it took some time to realize that the root cause is a change in console printing behavior in our application. As we want to track all errors that occur in production we use sink() to save all session console output, but unknowingly part of this implementation leaked in our test suit and stopped the following line from working correctly:
`if (any(grepl("Listening on http", err_lines))) break`

**Comment by hadley** [comment]
I think probably something is going wrong with app init, so the primary fix is to do better when startup fails: #329.
